### ISSUE TITLE
Medaka stitch concurrent future error

### ISSUE THREAD

**Issue by esteinig** [nanoporetech/medaka#234]
**Describe the bug**

Medaka fails unpredictably during stitching due to `BrokenProcessPool` error

**Logging**

```
File "/opt/medaka/venv/lib/python3.6/site-packages/medaka-1.2.1-py3.6-linux-x86_64.egg/medaka/medaka.py", line 708, in haploid_variant_cmdline
    medaka.wrappers.haploid_variant(args)
  File "/opt/medaka/venv/lib/python3.6/site-packages/medaka-1.2.1-py3.6-linux-x86_64.egg/medaka/wrappers.py", line 130, in haploid_variant
    medaka.stitch.stitch(args)
  File "/opt/medaka/venv/lib/python3.6/site-packages/medaka-1.2.1-py3.6-linux-x86_64.egg/medaka/stitch.py", line 182, in stitch
    contigs, gt = fill_gaps(contigs, args.draft)
  File "/opt/medaka/venv/lib/python3.6/site-packages/medaka-1.2.1-py3.6-linux-x86_64.egg/medaka/stitch.py", line 92, in fill_gaps
    for info, sequence_parts in contigs:
  File "/opt/medaka/venv/lib/python3.6/site-packages/medaka-1.2.1-py3.6-linux-x86_64.egg/medaka/stitch.py", line 130, in collapse_neighbours
    contig = next(contigs)
  File "/usr/lib/python3.6/concurrent/futures/process.py", line 366, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/lib/python3.6/concurrent/futures/_base.py", line 586, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.6/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
```

**Environment (if you do not have a GPU, write No GPU):**

 - Installation method from BioConda v1.2.1 in Docker container running in Singularity or Docker
 - OS: Ubuntu 18.04 (`nvidia/cuda:10.2-cudnn8-devel-ubuntu18.04` or `ubuntu:18.04`)
 - no GPU 

**Additional context**

Occurs on different systems (HPC nodes, GPU servers, CPU servers), across different runs (usually a single or few failures across hundreds of isolates) with different samples across runs affected (not predictable); occurs both in Singularity / Docker executions with or without Nextflow

**Comment by cjw85** [comment]
I suspect this might be related to https://github.com/nanoporetech/medaka/issues/246

**Comment by cjw85** [comment]
Hi @esteinig,

We have made a release candidate version of medaka with some fixes, would you mind trying it to see if you still have problems:

    pip install medaka==1.2.2rc1

**Comment by mlosilla** [comment]
Hi,

I tried v1.2.2rc1, but got the same error I had in v1.2.1. My error looks more like the one in issue #246. 

Traceback (most recent call last):
  File "<path>/lib/python3.8/site-packages/medaka/stitch.py", line 130, in collapse_neighbours
    contig = next(contigs)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<path>/bin/medaka", line 8, in <module>
    sys.exit(main())
  File "<path>/lib/python3.8/site-packages/medaka/medaka.py", line 683, in main
    args.func(args)
  File "<path>/lib/python3.8/site-packages/medaka/stitch.py", line 192, in stitch
    for (rname, start, stop), seq_parts in contigs:
RuntimeError: generator raised StopIteration

**Comment by mlosilla** [comment]
I have more info on the error I get. I noticed that the stitch module did run under haploid_variant. Long story short, I think my problem is caused by the file "consensus.hdf" produced by smolecule when using the flag --check_output (this file is an input for stitch). 

I compared both consensus.hdf files generated by smolecule with and without the --check_output flag:

h5dump good_consensus.hdf > good
h5dump bad_consensus.hdf > bad
diff good bad

This is what I got: 

1c1
< HDF5 "good_consensus.hdf" {
---
> HDF5 "bad_consensus.hdf" {
56011c56011
<             STRSIZE 208;
---
>             STRSIZE 5;
56018c56018
<          (0): "\37777777600\004\37777777625\37777777705\000\000\000\000\000\000\000\37777777617\37777777624(\37777777614\024250reads:496.3-953.1\37777777624\37777777614\026250reads:953.2-1316.12\37777777624\37777777614\027250reads:1316.13-1691.0\37777777624\37777777614\026250reads:1449.2-1922.0\37777777624\37777777614\026250reads:1192.0-1457.0\37777777624\37777777614\025250reads:721.0-1191.0\37777777624\37777777614\024250reads:266.2-720.1\37777777624\37777777614\022250reads:0.0-496.2\37777777624\37777777620."
---
>          (0): "\37777777600\004\37777777617\37777777624."


I hope it helps. 

FWIW, every time I run smolecule the first output line says: "Given 3 input files, assuming one read per file.". This is incorrect, I only provide one input. I think it is interpreting 'output' and 'my_output_folder' as additional inputs. My smolecule command is:
medaka smolecule fasta my_input_file  output my_output_folder

Mau
